### ISSUE TITLE
how to run language_tool_python.tool("en-US") from pre-downloaded server

### ISSUE THREAD

**Issue by pvester** [jxmorris12/language_tool_python#5]
Hey,

I need to run the language_tool_python package on a remote server that does not allow for the tool to download the "en-US" server (URLError) when its running there. I can do the pip install language_tool_python without problems but can not download the server during runtime. Is there a way that I can manually download the "en-Us" server, copy it to the remote server and run language_tool_python from there without connecting to the internet when i run the language_tool_python.tool ?

**Comment by jxmorris12** [comment]
Hi @pvester, I can definitely help you get this system set up. I'm confused about the `URLError` you're seeing. Does the remote server not have access to the web?

**Comment by pvester** [comment]
thanks ! :) the remote server does not have access to download files from many different places, including apparently the "en-US" server in runtime. the error is something like: urllib.error.URLError: (urlopen error Remote and closed connection without response

**Comment by pvester** [comment]
I was thinking of doing something like this:
1) download the LanguageTool-4.9 manually
2) copy it to the remote server
3) setup a localhost on the remote server to run the languagetool with java 
4) use languageTool('en-US', remote_server = "")

you think it would work?

**Comment by jxmorris12** [comment]
Hey @pvester - I've realized there are two issues here:

1. For some reason, some systems get a `URLError` when trying to download from the LT server
2. There's no option to specify where to find LT. Right now, it downloads to the folder that the package is in (which probably isn't ideal anyway).

I'll take a crack at fixing both of these and let you know when there's a new version to try. Should be later today!

**Comment by jxmorris12** [comment]
Hi again @pvester --

I updated the package! Try reinstalling (`pip install language_tool_python --upgrade`) to try the latest changes. Both fixes should help you.

1. We use a different package for downloading now (`requests`) so, hopefully, you shouldn't get the `URLError` anymore on the other system
2. If you still do, you can download LanguageTool locally and `scp` it to a folder on the remote server, then use set `LTP_PATH` to point to wherever you saved `LanguageTool-*/`.

Hopefully, one of these will fix your issue. Please let me know!

**Comment by pvester** [comment]
Thanks alot! That was fast. I will try it out ASAP but today there is some maintenance of the server. I hope I will be able to try it out tomorrow.

**Comment by pvester** [comment]
Hey 
1) did not work, ProxyError. HTTPSConnectionPool
2) I am not sure how to rest til LTP_PATH, which command? I tried to use LTP_PATH = LanguageTool-*/ in bash, but not sure that command works as intended

**Comment by jxmorris12** [comment]
Hey @pvester, sorry about the ProxyError -- not sure how or why that occurred.

about the path, let me be more specific:

1. Move LanguageTool over to some folder. Let's say I moved it to `/home/jxmorris12/stuff/`, so the full path of my LanguageTool is `/home/jxmorris12/stuff/LanguageTool-4.9/`.
2. Set `LTP_PATH` to the folder containing language tool. It's an environment variable, so the easiest way to set it is using `export` on Unix: `export LTP_PATH=/home/jxmorris12/stuff`. Once running the `export` command you should be able to use `language_tool_python` like normal.

**Comment by pvester** [comment]
I tried doing that and get HTTP Error 502: Connection Refured in server.py + keyError 'languages' in utils.py.

I did however manage to get it to work PERFECTLY by
1. Copying the LanguageTool-4.9 to the remote server
2. Setting up a Java HTTP Server based on LanguageTool-4.9 running on port 8080
3. changing the python_language_tool.LanguageTool('en-US') to python_language_tool.LanguageTool('en-US', remote_server = "http://localhost:8080")
